src/color-mode/color-mode.module.ts

```diff
@@ -0,0 +1,46 @@
+import { ModuleWithProviders, NgModule } from '@angular/core';
+
+import { provideColorMode } from './color-mode.provider';
+import { ColorModeConfig } from './color-mode.service';
+
+/**
+ * The `ColorModeModule` is an Angular module that provides color mode configuration and services.
+ *
+ * This module can be imported into your Angular application to enable color mode functionality.
+ *
+ * @example
+ * // Importing the module in your application
+ * import { ColorModeModule } from './color-mode/color-mode.module';
+ *
+ * @NgModule({
+ *   imports: [
+ *     ColorModeModule,
+ *   ],
+ * })
+ *
+ * or with configurations
+ *
+ * @NgModule({
+ *   imports: [
+ *     ColorModeModule.forRoot({
+ *       	defaultColorMode: 'auto',
+ *				saveToLocalStorage: true,
+ *				localStorageKey: 'bs_preferred_color_mode',
+ *     }),
+ *   ],
+ * })
+ * export class AppModule {}
+ *
+ * @module
+ */
+@NgModule({
+	providers: provideColorMode(),
+})
+export class ColorModeModule {
```

Comment:
We shouldn't use modules anymore. The provide function can be used whether or not modules are used in the application.